cabal2nix/src/Distribution/Nixpkgs/Haskell/FromCabal/License.hs

```diff
@@ -9,24 +9,25 @@ module Distribution.Nixpkgs.Haskell.FromCabal.License
 import Data.List (intercalate)
 import Distribution.License ( License(..), knownLicenses )
 import Distribution.Nixpkgs.License
-import Distribution.Pretty (prettyShow)
+import qualified Distribution.Pretty as DPretty
+import Language.Nix.PrettyPrinting (prettyShow)
 import qualified Distribution.SPDX as SPDX
 import Distribution.Text (display)
 import Distribution.Version
 
 -- TODO: Programmatically strip trailing zeros from license version numbers.
 
 fromCabalLicense :: Distribution.License.License -> Distribution.Nixpkgs.License.License
-fromCabalLicense (GPL Nothing)                             = Unknown (Just "GPL")
+fromCabalLicense (GPL Nothing)                             = Known "lib.licenses.gpl2Plus"
```

Comment:
These cases are intentional as there is not enough information to conclude what variant of the license in question is used.

Reply:
I do realize they were intentional, but they can't stay in their current form. I'm currently working towards disallowing stringy licenses in nixpkgs, and those are pretty much the only ones left. Other options would be to make it even less informational with `lib.licenses.free` or to add a new custom license (not really a fan of this).